lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp

```diff
@@ -803,37 +805,28 @@ void ProxyRenderDelegate::_UpdateRenderTags()
         // Build the list of render tags which were added or removed (changed)
         // and the list of render tags which were removed.
         TfTokenVector changedRenderTags;
-        TfTokenVector removedRenderTags;
         if (renderPurposeChanged) {
             changedRenderTags.push_back(HdRenderTagTokens->render);
-            if (!_proxyShapeData->DrawRenderPurpose())
-                removedRenderTags.push_back(HdRenderTagTokens->render);
         }
         if (proxyPurposeChanged) {
             changedRenderTags.push_back(HdRenderTagTokens->proxy);
-            if (!_proxyShapeData->DrawProxyPurpose())
-                removedRenderTags.push_back(HdRenderTagTokens->proxy);
         }
         if (guidePurposeChanged) {
             changedRenderTags.push_back(HdRenderTagTokens->guide);
-            if (!_proxyShapeData->DrawGuidePurpose())
-                removedRenderTags.push_back(HdRenderTagTokens->guide);
         }
 
         // Mark all the rprims which have a render tag which changed dirty
         SdfPathVector    rprimsToDirty = _GetFilteredRprims(*_defaultCollection, changedRenderTags);
-        HdChangeTracker& changeTracker = _renderIndex->GetChangeTracker();
+        
         for (auto& id : rprimsToDirty) {
             changeTracker.MarkRprimDirty(id, HdChangeTracker::DirtyRenderTag);
         }
+    }
 
-        // Make a special pass over the removed render tags so that the objects
-        // can hide themselves.
-        _taskController->SetRenderTags(removedRenderTags);
-        _engine.Execute(_renderIndex.get(), &_dummyTasks);
-
-        // Set the new render tags correctly.The regular execute will cause
-        // any newly visible rprims will update and mark themselves visible.
+    // When the render tag on an rprim changes we do a pass over all rprims to update
+    // their visibility. The frame after we do the pass over all the tags, set the tags back to
+    // the minimum set of tags.
```

Comment:
In the new version we call Execute once with all the tags (to hide those visible->hidden rprims) and the next update we go back to calling Execute with the minimum set of render tags.